code/modules/clothing/shoes/magboots.dm

```diff
@@ -53,12 +53,24 @@
 	slowdown_active = SHOES_SLOWDOWN
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
 
+/obj/item/clothing/shoes/magboots/advance/attack_self(mob/user)
+	if(magpulse)
```

Comment:
never turns off?

Reply:
Pretty sure it should be handled fine since it's a subtype?